app/grandchallenge/components/form_fields.py

```diff
@@ -102,10 +103,19 @@ def get_initial_value(self):
             return self.initial
 
     def get_image_field(self):
+        current_value = None
+
+        if isinstance(self.initial, ComponentInterfaceValue):
+            current_value = self.initial.image
+        elif Image.objects.filter(pk=self.initial).exists():
+            current_value = Image.objects.filter(pk=self.initial).first()
+        elif UserUpload.objects.filter(pk=self.initial).exists():
+            current_value = UserUpload.objects.filter(pk=self.initial).first()
```

Comment:
We should handle cases other than these 3. 

And since you're checking for the existence of Image and UserUpload, why don't you just use get()?

Reply:
Done